src/dataset/impl_dataset.rs

```diff
@@ -123,7 +123,7 @@ impl<R: Records, S> DatasetBase<R, S> {
     }
 }
 
-impl<L, R: Records, T: AsTargets<Elem = L>> DatasetBase<R, T> {
+impl<L, R: Records, T: AsSingleTargets<Elem = L>> DatasetBase<R, T> {
```

Comment:
@YuhanLiin The implementation for `T: AsMultiTargets<Elem = L>` would be very similar. What do you recommend to avoid duplicating very similar code?

Reply:
Use `AsMultiTargets` instead of `AsSingleTarget` as the bound for these dataset impls. `AsMultiTarget` is more general than `AsSingleTarget`, and even without a blanket impl, types that implement `AsSingleTarget` probably also implement `AsMultiTarget`.